spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/mail/SesAutoConfiguration.java

```diff
@@ -0,0 +1,99 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.aws.autoconfigure.mail;
+
+import javax.mail.Session;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.simpleemail.AmazonSimpleEmailService;
+import com.amazonaws.services.simpleemail.AmazonSimpleEmailServiceClient;
+
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.mail.MailSenderAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.aws.autoconfigure.context.ContextCredentialsAutoConfiguration;
+import org.springframework.cloud.aws.context.annotation.ConditionalOnMissingAmazonClient;
+import org.springframework.cloud.aws.core.config.AmazonWebserviceClientFactoryBean;
+import org.springframework.cloud.aws.core.region.RegionProvider;
+import org.springframework.cloud.aws.core.region.StaticRegionProvider;
+import org.springframework.cloud.aws.ses.SimpleEmailServiceJavaMailSender;
+import org.springframework.cloud.aws.ses.SimpleEmailServiceMailSender;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.mail.MailSender;
+import org.springframework.mail.javamail.JavaMailSender;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for AWS Simple Email Service
+ * support.
+ *
+ * @author Agim Emruli
+ * @author Eddú Meléndez
+ */
+@Configuration(proxyBeanMethods = false)
+@AutoConfigureAfter(MailSenderAutoConfiguration.class)
+@ConditionalOnClass({ AmazonSimpleEmailService.class, MailSender.class })
+@ConditionalOnMissingBean(MailSender.class)
+@Import(ContextCredentialsAutoConfiguration.class)
+@EnableConfigurationProperties(SesProperties.class)
+@ConditionalOnProperty(name = "spring.cloud.aws.ses.enabled", havingValue = "true",
+		matchIfMissing = true)
+public class SesAutoConfiguration {
```

Comment:
What happens if both old mail auto-config and new SES auto-config are enabled? I cannot figure it out just by looking at the code. Is there an order somewhere baked in?

Reply:
good catch! I have added `@ConditionalOnMissingClass("org.springframework.cloud.aws.mail.simplemail.SimpleEmailServiceJavaMailSender")`

Reply:
Assuming that `context` at least now is always included in user's projects, this would mean that `SesAutoConfiguration` never gets triggered, right?

Maybe we can make it in a way, that `SesAutoConfiguration` has `@AutoConfigureBefore(SimpleEmailAutoConfiguration.class)`? Then in case user intentionally adds SES Starter, old `SimpleEmailAutoConfiguration` will not fire because the `MailSender` bean will be already defined. What do you think?

Reply:
> Assuming that `context` at least now is always included in user's projects, this would mean that `SesAutoConfiguration` never gets triggered, right?

That's right. At least during 2.3

I like the idea. Will do the changes later today 👍🏽

Reply:
changes were applied